Lib/test/libregrtest/main.py

```diff
@@ -208,11 +209,12 @@ def find_tests(self, tests: TestList | None = None) -> tuple[TestTuple, TestList
                 print(f"Cannot find starting test: {self.starting_test}")
                 sys.exit(1)
 
-        if self.randomize:
+        if self.use_random_seed:
             if self.random_seed is None:
                 self.random_seed = random.randrange(100_000_000)
```

Comment:
The seed must always be initialized, but shuffle() must only be called in self.randomize is true.

You can move seed initialization in _init().